src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/external/UserSearchController.kt

```diff
@@ -24,6 +30,56 @@ class UserSearchController(
   private val userSearchService: UserSearchService
 ) {
 
+  @GetMapping("/search")
+  @Operation(
+    summary = "Search for an external user."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  @PreAuthorize(
+    "hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')"
+  )
+  suspend fun searchForUser(
+    @Parameter(
+      description = "The username, email or name of the user.",
+      example = "j smith"
+    ) @RequestParam(required = false)
+    name: String?,
+    @Parameter(description = "The role codes of the user.") @RequestParam(required = false)
+    roles: List<String>?,
+    @Parameter(description = "The group codes of the user.") @RequestParam(required = false)
+    groups: List<String>?,
+    @Parameter(description = "Limit to active / inactive / show all users.") @RequestParam(
+      required = false,
+      defaultValue = "ALL"
```

Comment:
Not sure if we need status to be mandatory as external users sorts all this out for us? 
Probably just design preference so happy either way